contracts/gateway/OSTComposer.sol

```diff
@@ -0,0 +1,427 @@
+        uint256 bounty = gateway.bounty();
+        require(
+            baseToken.transferFrom(msg.sender, address(stakerProxy), bounty),
+            "Bounty amount must be transferred to stakerProxy."
+        );
+
+        messageHash_ = StakerProxy(stakerProxy).stake(
+            stakeRequest.amount,
+            stakeRequest.beneficiary,
+            stakeRequest.gasPrice,
+            stakeRequest.gasLimit,
+            stakeRequest.nonce,
+            _hashLock,
+            stakeRequest.gateway
+        );
+
+        delete stakeRequests[_stakeRequestHash];
+        delete stakeRequestHashes[stakeRequest.staker][stakeRequest.gateway];
+    }
+
+    /**
+     * @notice It can only be called by Staker of the stake request. It deletes the
+     *         previously requested stake request.
+     *
+     * @param _stakeRequestHash Hash of the stake request.
+     *
+     * @return success_ `true` if stake request is deleted.
+     */
+    function revokeStakeRequest(
+        bytes32 _stakeRequestHash
+    )
+        external
+        returns(bool success_)
+    {
+        address staker = stakeRequests[_stakeRequestHash].staker;
+        require(
+            staker == address(0),
+            "Invalid stake request hash."
+        );
+        require(
+            staker == msg.sender,
+            "Only valid staker can revoke the stake request."
+        );
+
+        activeStakeRequestCount[msg.sender].sub(1);
+
+        address gateway = stakeRequests[_stakeRequestHash].gateway;
+        delete stakeRequestHashes[staker][gateway];
+        delete stakeRequests[_stakeRequestHash];
+
+        EIP20Interface token = EIP20GatewayInterface(gateway).valueToken();
+
+        uint256 amount = stakeRequests[_stakeRequestHash].amount;
+
+        require(
+            token.transfer(gateway, amount),
+            "Staked amount must be transferred to staker."
+        );
+
+        emit StakeRevoked(_stakeRequestHash, msg.sender);
+
+        success_ = true;
+    }
+
+    /**
+     * @notice It can only be called by Facilitator. It deletes the previously requested
+     *         stake request.
+     *
+     * @param _stakeRequestHash Hash of the stake request.
+     *
+     * @return success_ `true` if stake request is deleted.
+     */
+    function rejectStakeRequest(
+        bytes32 _stakeRequestHash
+    )
+        external
+        onlyWorker
+        returns(bool success_)
+    {
+        StakeRequest storage stakeRequest = stakeRequests[_stakeRequestHash];
+        require(stakeRequest.staker == address(0),"Invalid stake request hash.");
+
+        address staker = stakeRequests[_stakeRequestHash].staker;
+        activeStakeRequestCount[staker].sub(1);
+        address gateway = stakeRequests[_stakeRequestHash].gateway;
+
+        delete stakeRequests[_stakeRequestHash];
+        delete stakeRequestHashes[stakeRequest.staker][stakeRequest.gateway];
+
+        EIP20Interface token = EIP20GatewayInterface(gateway).valueToken();
```

Comment:
If you get the token before the deletes together with the gateway, it make more sense to the reader. Right now the gateway somewhat dangles mid-air up there.